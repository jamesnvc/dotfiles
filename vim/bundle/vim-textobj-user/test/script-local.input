let s:counter = [0]
function! s:increment()
  let s:counter[0] += 1
  return 0
endfunction

call textobj#user#plugin('dummy', {
\   '-': {
\     '*sfile*': expand('<sfile>:p'),
\     '*select-function*': 's:increment',
\     'select': ['ad', 'id'],
\   },
\ })

function s:describe__textobj_user_plugin__sfile()  "{{{1
  let counter = s:counter

  It should be able to call script-local function.

  Should counter[0] == 0

  execute 'normal' "vid\<Esc>"
  Should counter[0] == 1

  execute 'normal' "vad\<Esc>"
  Should counter[0] == 2
endfunction




" __END__  "{{{1
" vim: filetype=vim foldmethod=marker
