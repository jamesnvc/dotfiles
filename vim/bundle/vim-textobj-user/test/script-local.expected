==== textobj_user_plugin sfile
---- It should be able to call script-local function. 
...



**** Result ****


3 examples, 0 failures
